rpcs3/Emu/RSX/D3D12/D3D12GSRender.cpp

```diff
@@ -0,0 +1,1363 @@
+#include "stdafx.h"
+#if defined(DX12_SUPPORT)
+#include "D3D12GSRender.h"
+#include <wrl/client.h>
+#include <dxgi1_4.h>
+#include <d3dcompiler.h>
+#include <thread>
+#include <chrono>
+
+PFN_D3D12_CREATE_DEVICE wrapD3D12CreateDevice;
+PFN_D3D12_GET_DEBUG_INTERFACE wrapD3D12GetDebugInterface;
+PFN_D3D12_SERIALIZE_ROOT_SIGNATURE wrapD3D12SerializeRootSignature;
+
+static HMODULE D3D12Module;
+
+static void loadD3D12FunctionPointers()
+{
+	D3D12Module = LoadLibrary(L"d3d12.dll");
+	wrapD3D12CreateDevice = (PFN_D3D12_CREATE_DEVICE)GetProcAddress(D3D12Module, "D3D12CreateDevice");
+	wrapD3D12GetDebugInterface = (PFN_D3D12_GET_DEBUG_INTERFACE)GetProcAddress(D3D12Module, "D3D12GetDebugInterface");
+	wrapD3D12SerializeRootSignature = (PFN_D3D12_SERIALIZE_ROOT_SIGNATURE)GetProcAddress(D3D12Module, "D3D12SerializeRootSignature");
+}
+
+static void unloadD3D12FunctionPointers()
+{
+	FreeLibrary(D3D12Module);
+}
+
+GetGSFrameCb2 GetGSFrame = nullptr;
+
+void SetGetD3DGSFrameCallback(GetGSFrameCb2 value)
+{
+	GetGSFrame = value;
+}
+
+GarbageCollectionThread::GarbageCollectionThread()
+{
+	m_worker = std::thread([this]() {
```

Comment:
How this thread is terminated? I suppose it must me joined in the destructor since it captures `this`.